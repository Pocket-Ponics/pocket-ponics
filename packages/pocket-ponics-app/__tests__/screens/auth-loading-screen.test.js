import React from 'react'
import { shallow } from 'enzyme'
import AuthUtil from '../../util/auth-util'

import AuthLoadingScreen from '../../screens/auth-loading-screen'

jest.mock('../../util/auth-util')
jest.mock('expo-permissions')

describe('Auth Screen', () => {
	test('renders correctly', () => {
		const tree = shallow(
			<AuthLoadingScreen />
		).html()
		expect(tree).toMatchSnapshot()
	})

	test('navigates to new pages', () => {
		const navigation = {
			navigate: jest.fn()
		}

		AuthUtil.getAuthToken.mockImplementation((fn1, fn2) => {
			fn1()
			fn2()
		})

		shallow(
			<AuthLoadingScreen navigation={navigation}/>
		)

		expect(navigation.navigate).toHaveBeenCalledTimes(2)
	})
})