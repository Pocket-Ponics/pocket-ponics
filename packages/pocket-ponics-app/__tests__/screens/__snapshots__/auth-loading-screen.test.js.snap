// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Screen renders correctly 1`] = `"<View style=\\"flex:1\\"><View style=\\"flex:1;align-items:center;justify-content:center;padding-bottom:100px\\"><Image source=\\"1\\" style=\\"height:200px;resize-mode:contain\\"></Image><ActivityIndicator color=\\"#FFFFFF\\"></ActivityIndicator></View></View>"`;
